cwms-data-api/src/main/java/cwms/cda/data/dao/basin/BasinDao.java

```diff
@@ -0,0 +1,179 @@
+package cwms.cda.data.dao.basin;
+
+import cwms.cda.data.dao.JooqDao;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.basin.Basin;
+import cwms.cda.api.enums.UnitSystem;
+import cwms.cda.api.enums.Unit;
+import org.jooq.exception.DataAccessException;
+import usace.cwms.db.jooq.dao.CwmsDbBasinJooq;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.sql.ResultSet;
+import java.util.List;
+import java.util.ArrayList;
+
+import org.jooq.DSLContext;
+
+public class BasinDao extends JooqDao<Basin> {
+
+    public BasinDao(DSLContext dsl) {
+        super(dsl);
+    }
+
+    public List<Basin> getAllBasins(String unitSystem, String officeId) throws SQLException {
+        List<Basin> retVal = new ArrayList<>();
+        CwmsDbBasinJooq basinJooq = new CwmsDbBasinJooq();
+        String areaUnitIn = UnitSystem.EN.value().equals(unitSystem)
+                ? Unit.SQUARE_MILES.getValue() : Unit.SQUARE_KILOMETERS.getValue();
+        try {
+            connection(dsl, c -> {
```

Comment:
use connectionResult for getOne/getMulti

Reply:
fixed in latest commit